provider/types.go

```diff
@@ -0,0 +1,11 @@
+package provider
```

Comment:
Currently provider is used only inside platform/ecsv3, therefore it seems we don't have to place top-level of container-agent.
Is there any future plan to use this provider elsewhere?

Reply:
It was also planned to be used on the Kubernetes platform, but it seems that just the platform package is enough.
Move to the ecsv3 package.